flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -469,7 +469,9 @@ public void onBuffer(Buffer buffer, int sequenceNumber, int backlog) throws IOEx
 
 			if (notifyReceivedBarrier != null) {
 				receivedCheckpointId = notifyReceivedBarrier.getId();
-				listener.notifyBarrierReceived(notifyReceivedBarrier, channelInfo);
+				if (notifyReceivedBarrier.isCheckpoint()) {
```

Comment:
I guess the above `receivedCheckpointId = notifyReceivedBarrier.getId();` should also be placed inside this condition, because `receivedCheckpointId` only works for checkpoint now.

Reply:
But the condition works the other way around? If we bump/increase `receivedCheckpointId` here, it shouldn't matter. Spilling from input channels starts only if `lastRequestedCheckpointId` is bumped over the `receivedCheckpointId`.

Reply:
Yes, it should be no problem here. I forgot the truth that the savepoint id is consecutive incremental based on checkpoint id.

Reply:
> above receivedCheckpointId < lastRequestedCheckpointId might be invalid 

Yes, the intention was to invalidate this condition. 
So the next data buffer after a *savepoint* barrier doesn't notify the listener and doesn't go to `ChannelStateWriter`.

>  to make some unnecessary notifyReceivedBuffer to be spilled in listener.

If the condition is invalidated then the listener is NOT notified.